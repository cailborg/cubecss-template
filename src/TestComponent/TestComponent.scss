@import "../scss/tokens";

.test {
  width: 400px;
  --flow-space: 3rem;
  &[data-variant="primary"] {
    color: red;
  }
  // We want this background colour to be configurable by the user - so it's specificity takes precedence over the utility class `background-` but is overriden by adding `--colors-primary` to the application root
  background-color: rgba(var(--colors-primary), 0.5);
}
