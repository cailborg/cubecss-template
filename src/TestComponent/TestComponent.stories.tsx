import React from "react";
import { Story, Meta } from "@storybook/react/types-6-0";
import { TestComponent, TestComponentProps } from "./TestComponent";

export default {
  title: "Components/TestComponent",
  component: TestComponent,
  argTypes: {
    children: { control: "text" },
  },
} as Meta;

const Template: Story<TestComponentProps> = (args) => <TestComponent {...args} />;

export const Standard = Template.bind({});
Standard.args = {
  children: "Button text",
};
