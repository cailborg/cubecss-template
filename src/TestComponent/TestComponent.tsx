import React, { ReactNode } from "react";
import { Flow, Wrapper } from "../Utilities/Utilities";
import "./TestComponent.scss"

export interface TestComponentProps {
  variant: "primary" | "secondary";
  children?: ReactNode;
}

export const TestComponent: React.FC<TestComponentProps> = ({variant="primary", children}) => {

  return (
    <Wrapper>
      <Flow classNames="group">
        <div className="ring-4 ring-amethyst-600 ring-offset-4 ring-offset-neutral-white rounded-sm transition-transform transform hover:translate-x-xl test placeholder-persimmon-100::-webkit-input-placeholder bg-primary px-tiny sm:px-5 m--1" data-variant={variant}>{children}</div>
        <div className=" test shadow-high border-thick border-eminence-600 hover:border-sky-600 border-indigo-600 p-jumbo hover:text-indigo-600 text-heading-xl group-hover:bg-green leading-heading-xl">{children}</div>
      </Flow>
    </Wrapper>
  );
};
