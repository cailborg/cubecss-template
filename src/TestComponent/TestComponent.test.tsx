import React from "react";
import { TestComponent } from "../TestComponent/TestComponent";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<TestComponent variant="primary"/>).toJSON();
  expect(tree).toMatchSnapshot();
});
