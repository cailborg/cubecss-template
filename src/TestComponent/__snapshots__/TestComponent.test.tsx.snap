// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="wrapper "
>
  <div
    className="flow "
  >
    <div
      className="test bg-secondary"
      data-variant="primary"
    />
    <div
      className="test bg-primary"
      data-variant="primary"
    />
  </div>
</div>
`;
